File: rb/features.rb
# Finsignals SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module FinsignalsFeatures
  def self.make_feature(name)
    case name
    when "base"
      FinsignalsBaseFeature.new
    when "ratelimit"
      FinsignalsRatelimitFeature.new
    when "retry"
      FinsignalsRetryFeature.new
    when "test"
      FinsignalsTestFeature.new
    when "timeout"
      FinsignalsTimeoutFeature.new
    else
      FinsignalsBaseFeature.new
    end
  end
end

File: py/features.py
# Finsignals SDK feature factory

from feature.base_feature import FinsignalsBaseFeature
from feature.test_feature import FinsignalsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FinsignalsBaseFeature(),
        "test": lambda: FinsignalsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
